components/RevisionTimeline/MiniSubmissionTimeline.spec.tsx

```diff
@@ -1,60 +1,133 @@
-import { render, screen } from '@testing-library/react';
-import { mockSubmission } from 'factories/submissions';
+import { render, screen, waitFor } from '@testing-library/react';
+import { Revision, Submission } from 'data/flexibleForms/forms.types';
+import {
+  mockInProgressSubmission,
+  mockSubmission,
+} from 'factories/submissions';
 import { mockedWorker } from 'factories/workers';
 import MiniRevisionTimeline from './MiniRevisionTimeline';
+import * as submissionHooks from 'utils/api/submissions';
+import { SWRResponse } from 'swr';
+import { ErrorAPI } from 'types';
+
+jest.mock('utils/api/submissions');
+jest.mock('components/Spinner/Spinner', () => () => 'MockedSpinner');
 
 describe('MiniRevisionTimeline', () => {
-  it('correctly renders edits and creation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('correctly renders edits and creation', async () => {
+    jest.spyOn(submissionHooks, 'useSubmission').mockImplementation(() => {
+      const response = {
+        data: mockSubmission,
+        error: undefined,
+        isValidating: false,
+      } as SWRResponse<Submission, ErrorAPI>;
+
+      return response;
+    });
+
     render(
-      <MiniRevisionTimeline
-        submission={{
-          ...mockSubmission,
-          editHistory: [
-            {
-              worker: mockedWorker,
-              editTime: '2021-06-23T12:00:00.000Z',
-            },
-            {
-              worker: mockedWorker,
-              editTime: '2021-07-28T11:00:00.000Z',
-            },
-          ],
-        }}
-      />
+      <MiniRevisionTimeline inProgressSubmission={mockInProgressSubmission} />
     );
 
-    expect(screen.queryAllByText('foo.bar@hackney.gov.uk').length).toBe(2);
-    expect(screen.getByText('28 Jul 2021', { exact: false }));
-    expect(screen.getByText('23 Jun 2021', { exact: false }));
+    await waitFor(() => {
+      expect(screen.getByText('foo.bar@hackney.gov.uk', { exact: false }));
+      expect(screen.getByText('21 Jun 2021', { exact: false }));
+    });
   });
 
-  it('only shows up to three events', () => {
+  it('only shows up to three events', async () => {
+    const mockSubmissionClone = JSON.parse(
+      JSON.stringify(mockSubmission)
+    ) as Submission;
```

Comment:
What's going on here – stringify then parse? 🤔

Reply:
Ah, I see you're cloning here. Normally our fixtures use the `.build()` pattern so you can create a fresh one at the start of the test, rather than having to worry about mutation issues.

Could we take a look at doing the same thing here?

Reply:
Yeah, I'll make this change, messy way to clone instead of using build